mailet/crypto/src/main/java/org/apache/james/transport/mailets/SMIMEDecrypt.java

```diff
@@ -178,6 +179,6 @@ public void service(Mail mail) throws MessagingException {
     }
 
     private String text(Part mimePart) throws IOException, MessagingException {
-        return IOUtils.toString(mimePart.getDataHandler().getInputStream());
+        return IOUtils.toString(mimePart.getDataHandler().getInputStream(), StandardCharsets.UTF_8);
```

Comment:
I guess it is more complicated here... Are you sure the mimepart is in UTF-8 ?

Please create at least an Apache ticket for it. (I know javax.mail API almost makes it impossible to solve this)

Reply:
this `text` method is called by the following line:
```
newMessage.setText(text(strippedMessage), Charsets.UTF_8.name());
```
so I'm just enforcing something that is already done